Calculating BLEU score for pretrained BLIP model

In [2]:
import pandas as pd
from transformers import BlipProcessor, BlipForConditionalGeneration
import torch
from PIL import Image
from tqdm import tqdm

# Load the model and processor from the local cache
processor = BlipProcessor.from_pretrained('./blip-image-captioning-base')
model = BlipForConditionalGeneration.from_pretrained('./blip-image-captioning-base').to("cuda")

# Load the CSV file
df = pd.read_csv('bleu_score_images.csv')

# Generate descriptions for each image with a progress bar
new_descriptions = []

for idx, row in tqdm(df.iterrows(), total=len(df), desc="Generating Descriptions"):
    image_path = row['image_path']
    image = Image.open(image_path).convert("RGB")
    
    inputs = processor(images=image, return_tensors="pt").to("cuda")
    out = model.generate(**inputs)
    new_description = processor.decode(out[0], skip_special_tokens=True)
    new_descriptions.append(new_description)

# Add a new column with the new descriptions
df['blip_description'] = new_descriptions

# Save the updated dataframe to a new CSV file
df.to_csv('blip_pretrained_bleu.csv', index=False)

# # Print the new descriptions
# for idx, desc in enumerate(new_descriptions):
#     print(f"Frame {idx+1}: {desc}")

Generating Descriptions:   0%|          | 0/495 [00:00<?, ?it/s]C:\Users\jaivi\anaconda3\envs\tf-gpunew\lib\site-packages\transformers\generation\utils.py:1353: UserWarning: Using `max_length`'s default (20) to control the generation length. This behaviour is deprecated and will be removed from the config in v5 of Transformers -- we recommend using `max_new_tokens` to control the maximum length of the generation.
  warnings.warn(
Generating Descriptions: 100%|██████████| 495/495 [01:50<00:00,  4.47it/s]


In [3]:
df.head()

image_path  \
0  bleu_frames\1039117223_frame_1.jpg   
1  bleu_frames\1039117223_frame_2.jpg   
2  bleu_frames\1039117223_frame_3.jpg   
3  bleu_frames\1039117223_frame_4.jpg   
4  bleu_frames\1039117223_frame_5.jpg   

                                         description  \
0  Miracle of nature - flowering oleanders in the...   
1  Miracle of nature - flowering oleanders in the...   
2  Miracle of nature - flowering oleanders in the...   
3  Miracle of nature - flowering oleanders in the...   
4  Miracle of nature - flowering oleanders in the...   

             blip_description  
0  pink flowers on olive tree  
1  pink flowers on olive tree  
2  pink flowers in the garden  
3  pink flowers in the garden  
4      pink flowers on a bush

In [4]:
import pandas as pd
from nltk.translate.bleu_score import sentence_bleu, SmoothingFunction
from tqdm import tqdm

# Load the CSV file
df = pd.read_csv('blip_pretrained_bleu.csv')

# Calculate BLEU scores with a progress bar
bleu_scores = []
smooth = SmoothingFunction().method4

for idx, row in tqdm(df.iterrows(), total=len(df), desc="Calculating BLEU Scores"):
    reference = row['description']
    candidate = row['blip_description']
    
    # Calculate BLEU score
    bleu_score = sentence_bleu([reference.split()], candidate.split(), smoothing_function=smooth)
    bleu_scores.append(bleu_score)

# Add the BLEU scores to the dataframe
df['bleu_score'] = bleu_scores

# Calculate the average BLEU score
average_bleu_score = sum(bleu_scores) / len(bleu_scores)

# Save the updated dataframe to a new CSV file
df.to_csv('blip_pretrained_bleu_with_scores.csv', index=False)

# Print the BLEU scores and the average BLEU score
for idx, score in enumerate(bleu_scores):
    print(f"Frame {idx+1}: BLEU Score = {score:.4f}")

print(f"Average BLEU Score = {average_bleu_score:.4f}")

Calculating BLEU Scores: 100%|██████████| 495/495 [00:00<00:00, 6612.08it/s]

Frame 1: BLEU Score = 0.0000
Frame 2: BLEU Score = 0.0000
Frame 3: BLEU Score = 0.0018
Frame 4: BLEU Score = 0.0018
Frame 5: BLEU Score = 0.0000
Frame 6: BLEU Score = 0.0286
Frame 7: BLEU Score = 0.0570
Frame 8: BLEU Score = 0.0570
Frame 9: BLEU Score = 0.0235
Frame 10: BLEU Score = 0.0235
Frame 11: BLEU Score = 0.1212
Frame 12: BLEU Score = 0.0000
Frame 13: BLEU Score = 0.0000
Frame 14: BLEU Score = 0.0000
Frame 15: BLEU Score = 0.0000
Frame 16: BLEU Score = 0.0153
Frame 17: BLEU Score = 0.0196
Frame 18: BLEU Score = 0.0153
Frame 19: BLEU Score = 0.0196
Frame 20: BLEU Score = 0.0153
Frame 21: BLEU Score = 0.0279
Frame 22: BLEU Score = 0.0136
Frame 23: BLEU Score = 0.0179
Frame 24: BLEU Score = 0.0251
Frame 25: BLEU Score = 0.0221
Frame 26: BLEU Score = 0.0046
Frame 27: BLEU Score = 0.0079
Frame 28: BLEU Score = 0.0069
Frame 29: BLEU Score = 0.0356
Frame 30: BLEU Score = 0.0356
Frame 31: BLEU Score = 0.0277
Frame 32: BLEU Score = 0.2907
Frame 33: BLEU Score = 0.2907
Frame 34: BLEU Scor

Calculating BLEU score for fine tuned BLIP model

In [5]:
import pandas as pd
from transformers import BlipProcessor, BlipForConditionalGeneration
import torch
from PIL import Image
from tqdm import tqdm

# Load the model and processor from the local cache
processor = BlipProcessor.from_pretrained('./trained_model')
model = BlipForConditionalGeneration.from_pretrained('./trained_model').to("cuda")

# Load the CSV file
df = pd.read_csv('bleu_score_images.csv')

# Generate descriptions for each image with a progress bar
new_descriptions = []

for idx, row in tqdm(df.iterrows(), total=len(df), desc="Generating Descriptions"):
    image_path = row['image_path']
    image = Image.open(image_path).convert("RGB")
    
    inputs = processor(images=image, return_tensors="pt").to("cuda")
    out = model.generate(**inputs)
    new_description = processor.decode(out[0], skip_special_tokens=True)
    new_descriptions.append(new_description)

# Add a new column with the new descriptions
df['blip_description'] = new_descriptions

# Save the updated dataframe to a new CSV file
df.to_csv('blip_finetuned_bleu.csv', index=False)

# # Print the new descriptions
# for idx, desc in enumerate(new_descriptions):
#     print(f"Frame {idx+1}: {desc}")

Some weights of BlipForConditionalGeneration were not initialized from the model checkpoint at ./trained_model and are newly initialized: ['text_decoder.bert.embeddings.position_ids']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Generating Descriptions:   0%|          | 0/495 [00:00<?, ?it/s]C:\Users\jaivi\anaconda3\envs\tf-gpunew\lib\site-packages\transformers\generation\utils.py:1353: UserWarning: Using `max_length`'s default (20) to control the generation length. This behaviour is deprecated and will be removed from the config in v5 of Transformers -- we recommend using `max_new_tokens` to control the maximum length of the generation.
  warnings.warn(
Generating Descriptions: 100%|██████████| 495/495 [02:16<00:00,  3.62it/s]


In [6]:
import pandas as pd
from nltk.translate.bleu_score import sentence_bleu, SmoothingFunction
from tqdm import tqdm

# Load the CSV file
df = pd.read_csv('blip_finetuned_bleu.csv')

# Calculate BLEU scores with a progress bar
bleu_scores = []
smooth = SmoothingFunction().method4

for idx, row in tqdm(df.iterrows(), total=len(df), desc="Calculating BLEU Scores"):
    reference = row['description']
    candidate = row['blip_description']
    
    # Calculate BLEU score
    bleu_score = sentence_bleu([reference.split()], candidate.split(), smoothing_function=smooth)
    bleu_scores.append(bleu_score)

# Add the BLEU scores to the dataframe
df['bleu_score'] = bleu_scores

# Calculate the average BLEU score
average_bleu_score = sum(bleu_scores) / len(bleu_scores)

# Save the updated dataframe to a new CSV file
df.to_csv('blip_finetuned_bleu_with_scores.csv', index=False)

# Print the BLEU scores and the average BLEU score
for idx, score in enumerate(bleu_scores):
    print(f"Frame {idx+1}: BLEU Score = {score:.4f}")

print(f"Average BLEU Score = {average_bleu_score:.4f}")

Calculating BLEU Scores: 100%|██████████| 495/495 [00:00<00:00, 6363.17it/s]

Frame 1: BLEU Score = 0.0054
Frame 2: BLEU Score = 0.0054
Frame 3: BLEU Score = 0.0054
Frame 4: BLEU Score = 0.0054
Frame 5: BLEU Score = 0.0000
Frame 6: BLEU Score = 0.0000
Frame 7: BLEU Score = 0.0000
Frame 8: BLEU Score = 0.0000
Frame 9: BLEU Score = 0.0000
Frame 10: BLEU Score = 0.0000
Frame 11: BLEU Score = 0.0157
Frame 12: BLEU Score = 0.0157
Frame 13: BLEU Score = 0.0187
Frame 14: BLEU Score = 0.0157
Frame 15: BLEU Score = 0.0157
Frame 16: BLEU Score = 0.0000
Frame 17: BLEU Score = 0.0176
Frame 18: BLEU Score = 0.0176
Frame 19: BLEU Score = 0.0176
Frame 20: BLEU Score = 0.0176
Frame 21: BLEU Score = 0.0000
Frame 22: BLEU Score = 0.1352
Frame 23: BLEU Score = 0.1352
Frame 24: BLEU Score = 0.0176
Frame 25: BLEU Score = 0.0000
Frame 26: BLEU Score = 0.0418
Frame 27: BLEU Score = 0.0102
Frame 28: BLEU Score = 0.0102
Frame 29: BLEU Score = 0.0418
Frame 30: BLEU Score = 0.0474
Frame 31: BLEU Score = 0.0279
Frame 32: BLEU Score = 0.0279
Frame 33: BLEU Score = 0.0279
Frame 34: BLEU Scor

The BLEU (Bilingual Evaluation Understudy) score is a metric used to evaluate the quality of text generated by a machine translation or text generation system by comparing it to one or more reference texts.

Range: BLEU scores range from 0 to 1, where 1 indicates a perfect match between the generated text and the reference text.

1. The BLEU score for pretrained BLIP model tested on 495 images(5 frames extracted from 99 videos) is 0.0205
2. The BLEU score for finetuned BLIP model tested on 495 images(5 frames extracted from 99 videos) is 0.0133

In [11]:
import pandas as pd
import requests
import json
import time  # Import time for adding delays
from tqdm import tqdm

# Load the CSV file
df = pd.read_csv('blip_pretrained_bleu.csv')

# Replace 'YOUR_API_KEY' with your actual API key
api_key = 'AIzaSyDXqUdgBCDZWyNDPvqaon0mytEttl6YEJI'
url = f'https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash-latest:generateContent?key={api_key}'

headers = {
    'Content-Type': 'application/json',
}

# Function to call the API and get the summary
def get_summary(combined_text):
    data = {
        'contents': [
            {
                'parts': [
                    {
                        'text': f'Summarize this text into a concise, meaningful single sentence without introductory phrases. The sentence should be a narration as if a blind person can understand what is happening. Be fact oriented rather than being poetic. Make it short:\n\n{combined_text}'
                    }
                ]
            }
        ]
    }
    try:
        response = requests.post(url, headers=headers, data=json.dumps(data))
        response.raise_for_status()  # Raise an HTTPError for bad responses (4xx or 5xx)
        response_json = response.json()
        # Print the response for debugging
        print(f"Response JSON: {response_json}")

        if 'candidates' in response_json and len(response_json['candidates']) > 0:
            summary = response_json['candidates'][0]['content']['parts'][0]['text']
        else:
            summary = "No summary available"
    except requests.exceptions.RequestException as e:
        # Print the error for debugging
        print(f"Request failed: {e}")
        summary = "API request failed"
    except KeyError as e:
        # Print the error for debugging
        print(f"KeyError: {e} in response JSON")
        summary = "Unexpected response format"

    return summary

# Initialize a list to store summaries
summaries = []

# Iterate through the DataFrame in batches of 5 frames with a progress bar
for i in tqdm(range(0, len(df), 5), desc="Generating Summaries"):
    descriptions = df['blip_description'].iloc[i:i+5].tolist()
    combined_text = " ".join(descriptions)
    summary = get_summary(combined_text)
    summaries.extend([summary] * 5)
    
    # Add a 1-second delay between each API call
    time.sleep(1)

    # Check if we have processed 25 batches and wait for 1 minute
    if (i // 5 + 1) % 10 == 0:
        print("Processed 10 batches. Waiting for 1 minute...")
        time.sleep(60)

# Add the summaries to the DataFrame
df['summary_description'] = summaries

# Save the updated DataFrame to a new CSV file
df.to_csv('blip_pretrained_bleu_with_summaries.csv', index=False)

# Print the summarized descriptions for the first few batches
for idx, summary in enumerate(summaries[::5]):  # Print one summary per batch
    print(f"Batch {idx+1} Summary: {summary}")

Generating Summaries:   0%|          | 0/99 [00:00<?, ?it/s]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Pink flowers are blooming on an olive tree, in the garden, and on a bush. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 70, 'candidatesTokenCount': 18, 'totalTokenCount': 88}}


Generating Summaries:   1%|          | 1/99 [00:01<02:47,  1.70s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A drone flies over a misty mountain forest in the morning, capturing the landscape from above. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 93, 'candidatesTokenCount': 18, 'totalTokenCount': 111}}


Generating Summaries:   2%|▏         | 2/99 [00:03<02:47,  1.73s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A bright, orange fire burns against a dark sky. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 11, 'totalTokenCount': 86}}


Generating Summaries:   3%|▎         | 3/99 [00:05<02:42,  1.69s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A white wall repeatedly features yellow flowers. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 87, 'candidatesTokenCount': 8, 'totalTokenCount': 95}}


Generating Summaries:   4%|▍         | 4/99 [00:06<02:38,  1.67s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A small fire burns in the darkness, sending smoke rising against the black backdrop. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 16, 'totalTokenCount': 91}}


Generating Summaries:   5%|▌         | 5/99 [00:08<02:38,  1.68s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man and woman walk across a wooden bridge over a lake as the sun sets. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 96, 'candidatesTokenCount': 17, 'totalTokenCount': 113}}


Generating Summaries:   6%|▌         | 6/99 [00:10<02:36,  1.68s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A hand holds a small metal device near a hard drive with a visible screw. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 89, 'candidatesTokenCount': 16, 'totalTokenCount': 105}}


Generating Summaries:   7%|▋         | 7/99 [00:11<02:36,  1.70s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Many people are walking down the street. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 90, 'candidatesTokenCount': 8, 'totalTokenCount': 98}}


Generating Summaries:   8%|▊         | 8/99 [00:13<02:30,  1.65s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man in an apron and tie prepares a drink, pours water into a bottle, and places a container filled with water near a fountain in a grassy area. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 92, 'candidatesTokenCount': 32, 'totalTokenCount': 124}}


Generating Summaries:   9%|▉         | 9/99 [00:15<02:33,  1.70s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man with a camera stands among tall, bamboo-like trees in a forest. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 86, 'candidatesTokenCount': 17, 'totalTokenCount': 103}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  10%|█         | 10/99 [01:16<29:59, 20.22s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A mosque sits in the middle of a lake, with the sun setting behind it. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 85, 'candidatesTokenCount': 17, 'totalTokenCount': 102}}


Generating Summaries:  11%|█         | 11/99 [01:18<21:23, 14.59s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The beach in Vari, Italy, is visible from above. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 105, 'candidatesTokenCount': 12, 'totalTokenCount': 117}}


Generating Summaries:  12%|█▏        | 12/99 [01:20<15:28, 10.67s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Chickens are confined in multiple enclosures. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 81, 'candidatesTokenCount': 8, 'totalTokenCount': 89}}


Generating Summaries:  13%|█▎        | 13/99 [01:22<11:22,  7.93s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman in a white bikini sits on a rock. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 91, 'candidatesTokenCount': 11, 'totalTokenCount': 102}}


Generating Summaries:  14%|█▍        | 14/99 [01:23<08:33,  6.04s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': "A woman's feet are visible in the corner of a room, and she sits on a couch with her legs crossed. \n"}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 107, 'candidatesTokenCount': 25, 'totalTokenCount': 132}}


Generating Summaries:  15%|█▌        | 15/99 [01:25<06:38,  4.75s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A rough surface holds papers with drawings of a clock, two darts, and a fish. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 86, 'candidatesTokenCount': 18, 'totalTokenCount': 104}}


Generating Summaries:  16%|█▌        | 16/99 [01:27<05:18,  3.84s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'An old church, possibly named "Holy", stands in the village of San de Oro, nestled among the Pyrenean mountains. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 97, 'candidatesTokenCount': 25, 'totalTokenCount': 122}}


Generating Summaries:  17%|█▋        | 17/99 [01:28<04:22,  3.21s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A human brain is enclosed in a transparent sphere, encircled by a wire. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 76, 'candidatesTokenCount': 15, 'totalTokenCount': 91}}


Generating Summaries:  18%|█▊        | 18/99 [01:30<03:41,  2.74s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man sits on a floating sail, while another man sails on the water with a sail, and two people, one a woman, are also in the water with a sail. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 94, 'candidatesTokenCount': 36, 'totalTokenCount': 130}}


Generating Summaries:  19%|█▉        | 19/99 [01:32<03:16,  2.46s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Black eggplants and black bean plants grow inside a greenhouse. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 78, 'candidatesTokenCount': 12, 'totalTokenCount': 90}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  20%|██        | 20/99 [02:34<26:38, 20.23s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Someone is painting, making two paper plates, and preparing a pizza on a table using a knife. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 83, 'candidatesTokenCount': 20, 'totalTokenCount': 103}}


Generating Summaries:  21%|██        | 21/99 [02:35<19:06, 14.69s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Two chessboards with pieces, one in front of and one behind Chinese flags. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 80, 'candidatesTokenCount': 16, 'totalTokenCount': 96}}


Generating Summaries:  22%|██▏       | 22/99 [02:37<13:51, 10.80s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A boat, a submarine, and a whale are in the ocean at sunset, with a constant humming sound. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 99, 'candidatesTokenCount': 22, 'totalTokenCount': 121}}


Generating Summaries:  23%|██▎       | 23/99 [02:39<10:13,  8.08s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman sits at a table, writing in a notebook and drinking coffee. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 84, 'candidatesTokenCount': 15, 'totalTokenCount': 99}}


Generating Summaries:  24%|██▍       | 24/99 [02:40<07:41,  6.15s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man, a woman with raised arms, and three other women are standing in doorways or windows, looking out. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 93, 'candidatesTokenCount': 23, 'totalTokenCount': 116}}


Generating Summaries:  25%|██▌       | 25/99 [02:42<05:58,  4.85s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Two adults are sitting on a couch, watching a man and a child drawing at a table. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 101, 'candidatesTokenCount': 19, 'totalTokenCount': 120}}


Generating Summaries:  26%|██▋       | 26/99 [02:44<04:46,  3.92s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Lava flows from Kilulu volcano into its crater. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 10, 'totalTokenCount': 85}}


Generating Summaries:  27%|██▋       | 27/99 [02:46<03:53,  3.24s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A city sprawls beneath the plane, with a tall building visible in the distance. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 91, 'candidatesTokenCount': 17, 'totalTokenCount': 108}}


Generating Summaries:  28%|██▊       | 28/99 [02:47<03:17,  2.78s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'People walk along a street with buildings and parked cars. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 98, 'candidatesTokenCount': 11, 'totalTokenCount': 109}}


Generating Summaries:  29%|██▉       | 29/99 [02:49<02:51,  2.45s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man wearing a red cap and headphones walks down the street and looks at his phone. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 95, 'candidatesTokenCount': 18, 'totalTokenCount': 113}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  30%|███       | 30/99 [03:51<23:15, 20.23s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A group of people stand together in a park, including a man and woman walking with a dog, and a woman holding a child. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 89, 'candidatesTokenCount': 27, 'totalTokenCount': 116}}


Generating Summaries:  31%|███▏      | 31/99 [03:52<16:38, 14.69s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Water is being poured from a bottle into a container, producing a dripping sound. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 88, 'candidatesTokenCount': 16, 'totalTokenCount': 104}}


Generating Summaries:  32%|███▏      | 32/99 [03:54<12:02, 10.79s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A house sits on the edge of a rocky beach, with a cliff rising behind it. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 97, 'candidatesTokenCount': 18, 'totalTokenCount': 115}}


Generating Summaries:  33%|███▎      | 33/99 [03:56<08:52,  8.06s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A bright blue light flashed in the night sky, illuminating a concert stage. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 82, 'candidatesTokenCount': 15, 'totalTokenCount': 97}}


Generating Summaries:  34%|███▍      | 34/99 [03:58<06:38,  6.13s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Many pigeons are eating food scattered on the street. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 10, 'totalTokenCount': 85}}


Generating Summaries:  35%|███▌      | 35/99 [03:59<05:06,  4.79s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Two men wearing hard hats and vests work in a factory, one using a cell phone and the other using a tablet and machine. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 100, 'candidatesTokenCount': 26, 'totalTokenCount': 126}}


Generating Summaries:  36%|███▋      | 36/99 [04:01<04:04,  3.88s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A house has a thick layer of snow on its roof. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 79, 'candidatesTokenCount': 12, 'totalTokenCount': 91}}


Generating Summaries:  37%|███▋      | 37/99 [04:03<03:19,  3.22s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The Boston skyline appears as the sun sets, casting a warm glow on the buildings. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 17, 'totalTokenCount': 92}}


Generating Summaries:  38%|███▊      | 38/99 [04:04<02:48,  2.76s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Many ducks are moving across the water. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 70, 'candidatesTokenCount': 8, 'totalTokenCount': 78}}


Generating Summaries:  39%|███▉      | 39/99 [04:06<02:26,  2.44s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': "A man sits at a desk, while two men in suits discuss business, footage shows a city street with a car driving on it, and the film's title is 'The Birth of the City'. \n"}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 95, 'candidatesTokenCount': 41, 'totalTokenCount': 136}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  40%|████      | 40/99 [05:08<19:56, 20.29s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Clouds are in the sky, over the ocean, and over a field. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 69, 'candidatesTokenCount': 15, 'totalTokenCount': 84}}


Generating Summaries:  41%|████▏     | 41/99 [05:10<14:13, 14.72s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A white border surrounds a black background, with a river running through a series of towns, one small, then a larger one, then two more small ones. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 87, 'candidatesTokenCount': 32, 'totalTokenCount': 119}}


Generating Summaries:  42%|████▏     | 42/99 [05:11<10:18, 10.85s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A person rides a mountain bike on a dirt path. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 90, 'candidatesTokenCount': 11, 'totalTokenCount': 101}}


Generating Summaries:  43%|████▎     | 43/99 [05:13<07:33,  8.10s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man walks in darkness, his path illuminated intermittently by a purple light. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 82, 'candidatesTokenCount': 15, 'totalTokenCount': 97}}


Generating Summaries:  44%|████▍     | 44/99 [05:15<05:39,  6.17s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Someone is sanding a wooden floor with a machine, wearing red boots and black shoes. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 91, 'candidatesTokenCount': 17, 'totalTokenCount': 108}}


Generating Summaries:  45%|████▌     | 45/99 [05:17<04:22,  4.86s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The sun disappears behind clouds as it sets over the water. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 85, 'candidatesTokenCount': 12, 'totalTokenCount': 97}}


Generating Summaries:  46%|████▋     | 46/99 [05:18<03:26,  3.89s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': "The National Bank of Bangladesh is located in Dhaka, and the Indian government's headquarters are in New Delhi. \n"}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 100, 'candidatesTokenCount': 22, 'totalTokenCount': 122}}


Generating Summaries:  47%|████▋     | 47/99 [05:20<02:49,  3.26s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Someone is repeatedly saying "sui" while listing sushi rolls containing shrimp, avocado, and salmon. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 96, 'candidatesTokenCount': 19, 'totalTokenCount': 115}}


Generating Summaries:  48%|████▊     | 48/99 [05:22<02:22,  2.79s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A waterway flows through a city, with buildings lining its banks and a bridge spanning its width. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 92, 'candidatesTokenCount': 19, 'totalTokenCount': 111}}


Generating Summaries:  49%|████▉     | 49/99 [05:24<02:04,  2.50s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A baby is lying on a bed. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 65, 'candidatesTokenCount': 8, 'totalTokenCount': 73}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  51%|█████     | 50/99 [06:25<16:31, 20.23s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man in the back of a bus is illuminated by a red light. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 92, 'candidatesTokenCount': 15, 'totalTokenCount': 107}}


Generating Summaries:  52%|█████▏    | 51/99 [06:27<11:43, 14.65s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'There are many red poppy flowers in the garden. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 10, 'totalTokenCount': 85}}


Generating Summaries:  53%|█████▎    | 52/99 [06:28<08:25, 10.75s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A small town nestled in the countryside with houses and roads spread out. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 90, 'candidatesTokenCount': 14, 'totalTokenCount': 104}}


Generating Summaries:  54%|█████▎    | 53/99 [06:30<06:10,  8.05s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The sky is dark, filled with countless bright points and a cloudy, milky band. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 78, 'candidatesTokenCount': 17, 'totalTokenCount': 95}}


Generating Summaries:  55%|█████▍    | 54/99 [06:32<04:36,  6.14s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Many green leaves in the forest are covered in water drops. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 70, 'candidatesTokenCount': 12, 'totalTokenCount': 82}}


Generating Summaries:  56%|█████▌    | 55/99 [06:33<03:30,  4.79s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman in a yoga pose is in front of a fireplace. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 106, 'candidatesTokenCount': 13, 'totalTokenCount': 119}}


Generating Summaries:  57%|█████▋    | 56/99 [06:35<02:46,  3.86s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A small red building stands in the center of a park. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 91, 'candidatesTokenCount': 12, 'totalTokenCount': 103}}


Generating Summaries:  58%|█████▊    | 57/99 [06:37<02:14,  3.19s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A swirling mix of blue and green fills the background. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 11, 'totalTokenCount': 86}}


Generating Summaries:  59%|█████▊    | 58/99 [06:39<01:52,  2.74s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A squirrel on the ground repeatedly makes a sharp, clicking sound. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 127, 'candidatesTokenCount': 13, 'totalTokenCount': 140}}


Generating Summaries:  60%|█████▉    | 59/99 [06:40<01:36,  2.42s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman carefully places colorful flowers into a vase. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 85, 'candidatesTokenCount': 10, 'totalTokenCount': 95}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  61%|██████    | 60/99 [07:42<13:07, 20.19s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Two people sit on a couch and eat. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 96, 'candidatesTokenCount': 9, 'totalTokenCount': 105}}


Generating Summaries:  62%|██████▏   | 61/99 [07:44<09:16, 14.64s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A boat rests on the water beside a mountain. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 94, 'candidatesTokenCount': 10, 'totalTokenCount': 104}}


Generating Summaries:  63%|██████▎   | 62/99 [07:45<06:38, 10.77s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man and woman walk through an open doorway into a house. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 88, 'candidatesTokenCount': 13, 'totalTokenCount': 101}}


Generating Summaries:  64%|██████▎   | 63/99 [07:47<04:49,  8.05s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Many people stand before a barrier and a raised platform. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 96, 'candidatesTokenCount': 11, 'totalTokenCount': 107}}


Generating Summaries:  65%|██████▍   | 64/99 [07:49<03:34,  6.13s/it]

Response JSON: {'candidates': [{'finishReason': 'SAFETY', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'MEDIUM'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 84, 'totalTokenCount': 84}}
KeyError: 'content' in response JSON


Generating Summaries:  66%|██████▌   | 65/99 [07:50<02:43,  4.82s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Five clouds are visible in the sky. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 70, 'candidatesTokenCount': 8, 'totalTokenCount': 78}}


Generating Summaries:  67%|██████▋   | 66/99 [07:52<02:07,  3.85s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Sand, water, and clouds. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 96, 'candidatesTokenCount': 7, 'totalTokenCount': 103}}


Generating Summaries:  68%|██████▊   | 67/99 [07:54<01:41,  3.17s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Water flows through a landscape, sometimes surrounded by trees and grass, sometimes through a forest, and sometimes across a field. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 86, 'candidatesTokenCount': 24, 'totalTokenCount': 110}}


Generating Summaries:  69%|██████▊   | 68/99 [07:55<01:25,  2.74s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A sequence of backgrounds changes from gold glitter to black and white, each with stars, followed by a gold background with many lights, ending with a light piercing the darkness. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 78, 'candidatesTokenCount': 34, 'totalTokenCount': 112}}


Generating Summaries:  70%|██████▉   | 69/99 [07:57<01:14,  2.47s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A field of tall green grass stretches out. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 70, 'candidatesTokenCount': 9, 'totalTokenCount': 79}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  71%|███████   | 70/99 [08:59<09:46, 20.23s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Two men are conversing in front of a green background. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 80, 'candidatesTokenCount': 11, 'totalTokenCount': 91}}


Generating Summaries:  72%|███████▏  | 71/99 [09:01<06:50, 14.67s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Water falls from a fountain in the park. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 9, 'totalTokenCount': 84}}


Generating Summaries:  73%|███████▎  | 72/99 [09:02<04:49, 10.73s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A cement block sits on a concrete floor, against a blue wall. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 95, 'candidatesTokenCount': 14, 'totalTokenCount': 109}}


Generating Summaries:  74%|███████▎  | 73/99 [09:04<03:28,  8.01s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Three women are celebrating Christmas, sitting on a couch, floor, and playing with their cell phones, while laughing. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 118, 'candidatesTokenCount': 23, 'totalTokenCount': 141}}


Generating Summaries:  75%|███████▍  | 74/99 [09:05<02:33,  6.13s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The logo features a white background with a black and white border and the word "cappuno" repeated several times. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 83, 'candidatesTokenCount': 23, 'totalTokenCount': 106}}


Generating Summaries:  76%|███████▌  | 75/99 [09:07<01:55,  4.82s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The text describes courtyards of three Moroccan mosques: Al-Khali Palace, Al-Qed Mosque, and Al-Har. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 120, 'candidatesTokenCount': 27, 'totalTokenCount': 147}}


Generating Summaries:  77%|███████▋  | 76/99 [09:09<01:28,  3.87s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'The word "thank" repeats endlessly against a green backdrop. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 75, 'candidatesTokenCount': 12, 'totalTokenCount': 87}}


Generating Summaries:  78%|███████▊  | 77/99 [09:11<01:10,  3.21s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman walks a dog on a leash alongside another woman pushing a stroller. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 88, 'candidatesTokenCount': 15, 'totalTokenCount': 103}}


Generating Summaries:  79%|███████▉  | 78/99 [09:12<00:58,  2.78s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'From atop a hill, a windmill stands in a grassy field overlooking a bay with sailboats. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 104, 'candidatesTokenCount': 19, 'totalTokenCount': 123}}


Generating Summaries:  80%|███████▉  | 79/99 [09:14<00:49,  2.47s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Ocean waves break against rocky shores. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 94, 'candidatesTokenCount': 7, 'totalTokenCount': 101}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  81%|████████  | 80/99 [10:16<06:24, 20.22s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Bright, flashing lights and moving cars illuminate a city at night. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 103, 'candidatesTokenCount': 13, 'totalTokenCount': 116}}


Generating Summaries:  82%|████████▏ | 81/99 [10:17<04:24, 14.68s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman in a wedding dress is adjusting her hair and dress. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 95, 'candidatesTokenCount': 13, 'totalTokenCount': 108}}


Generating Summaries:  83%|████████▎ | 82/99 [10:19<03:03, 10.79s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Cars drive on a street with buildings on both sides. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 86, 'candidatesTokenCount': 11, 'totalTokenCount': 97}}


Generating Summaries:  84%|████████▍ | 83/99 [10:21<02:08,  8.05s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A winding river flows through a dense forest, the setting sun casting a golden glow on the water. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 95, 'candidatesTokenCount': 20, 'totalTokenCount': 115}}


Generating Summaries:  85%|████████▍ | 84/99 [10:23<01:32,  6.15s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man walks down stairs from a building with a sign that says "Entrance," then passes a yellow building with a sign that says "Surf House." \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 100, 'candidatesTokenCount': 30, 'totalTokenCount': 130}}


Generating Summaries:  86%|████████▌ | 85/99 [10:24<01:07,  4.83s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Three glasses, one with champagne and a blurry background, another with champagne and bubbles, and a third with beer, all out of focus. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 82, 'candidatesTokenCount': 28, 'totalTokenCount': 110}}


Generating Summaries:  87%|████████▋ | 86/99 [10:26<00:51,  3.92s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A futuristic humanoid, a skeleton in a medical room, and a human skeleton are each situated in front of computer screens. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 88, 'candidatesTokenCount': 24, 'totalTokenCount': 112}}


Generating Summaries:  88%|████████▊ | 87/99 [10:28<00:39,  3.26s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A woman sits at a table with a laptop and a tablet. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 93, 'candidatesTokenCount': 13, 'totalTokenCount': 106}}


Generating Summaries:  89%|████████▉ | 88/99 [10:29<00:30,  2.75s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man in a white shirt is tying a tie, then fastens it with a clip, and then puts a gold ring on his finger. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 94, 'candidatesTokenCount': 29, 'totalTokenCount': 123}}


Generating Summaries:  90%|████████▉ | 89/99 [10:31<00:24,  2.46s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Hands spin clay on a rotating wheel, forming a pot. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 105, 'candidatesTokenCount': 12, 'totalTokenCount': 117}}
Processed 10 batches. Waiting for 1 minute...


Generating Summaries:  91%|█████████ | 90/99 [11:33<03:02, 20.22s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man, using his smartphone, moves from a bench to a street, then to a park, then back to the street twice. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 95, 'candidatesTokenCount': 27, 'totalTokenCount': 122}}


Generating Summaries:  92%|█████████▏| 91/99 [11:35<01:57, 14.71s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A blue card with a white border has the numbers 14, hearts, and the words "Valentine Day" repeated. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 81, 'candidatesTokenCount': 25, 'totalTokenCount': 106}}


Generating Summaries:  93%|█████████▎| 92/99 [11:36<01:15, 10.83s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'Cherry blossoms are blooming in abundance along a river in Kyoto, Japan. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 89, 'candidatesTokenCount': 14, 'totalTokenCount': 103}}


Generating Summaries:  94%|█████████▍| 93/99 [11:38<00:48,  8.08s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A large blue flower, green seaweed, a blue and yellow fish, a blue and green abstract background, and another blue and yellow flower are all seen up close. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 86, 'candidatesTokenCount': 33, 'totalTokenCount': 119}}


Generating Summaries:  95%|█████████▍| 94/99 [11:40<00:31,  6.21s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'From above, a multitude of towering structures dominate a vast urban landscape, with a train snaking through its heart. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 84, 'candidatesTokenCount': 23, 'totalTokenCount': 107}}


Generating Summaries:  96%|█████████▌| 95/99 [11:42<00:19,  4.87s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A cage holds many colorful balls. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 72, 'candidatesTokenCount': 7, 'totalTokenCount': 79}}


Generating Summaries:  97%|█████████▋| 96/99 [11:44<00:11,  3.97s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A man is working with metal and wood in a factory. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 92, 'candidatesTokenCount': 12, 'totalTokenCount': 104}}


Generating Summaries:  98%|█████████▊| 97/99 [11:45<00:06,  3.28s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A spider sits in its web. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 69, 'candidatesTokenCount': 7, 'totalTokenCount': 76}}


Generating Summaries:  99%|█████████▉| 98/99 [11:47<00:02,  2.78s/it]

Response JSON: {'candidates': [{'content': {'parts': [{'text': 'A boat approaches a pier extending into the ocean. \n'}], 'role': 'model'}, 'finishReason': 'STOP', 'index': 0, 'safetyRatings': [{'category': 'HARM_CATEGORY_SEXUALLY_EXPLICIT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HATE_SPEECH', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_HARASSMENT', 'probability': 'NEGLIGIBLE'}, {'category': 'HARM_CATEGORY_DANGEROUS_CONTENT', 'probability': 'NEGLIGIBLE'}]}], 'usageMetadata': {'promptTokenCount': 97, 'candidatesTokenCount': 10, 'totalTokenCount': 107}}


Generating Summaries: 100%|██████████| 99/99 [11:49<00:00,  7.16s/it]

Batch 1 Summary: Pink flowers are blooming on an olive tree, in the garden, and on a bush. 

Batch 2 Summary: A drone flies over a misty mountain forest in the morning, capturing the landscape from above. 

Batch 3 Summary: A bright, orange fire burns against a dark sky. 

Batch 4 Summary: A white wall repeatedly features yellow flowers. 

Batch 5 Summary: A small fire burns in the darkness, sending smoke rising against the black backdrop. 

Batch 6 Summary: A man and woman walk across a wooden bridge over a lake as the sun sets. 

Batch 7 Summary: A hand holds a small metal device near a hard drive with a visible screw. 

Batch 8 Summary: Many people are walking down the street. 

Batch 9 Summary: A man in an apron and tie prepares a drink, pours water into a bottle, and places a container filled with water near a fountain in a grassy area. 

Batch 10 Summary: A man with a camera stands among tall, bamboo-like trees in a forest. 

Batch 11 Summary: A mosque sits in the middle of a lak

In [17]:
df.head(20)

image_path  \
0   bleu_frames\1039117223_frame_1.jpg   
1   bleu_frames\1039117223_frame_2.jpg   
2   bleu_frames\1039117223_frame_3.jpg   
3   bleu_frames\1039117223_frame_4.jpg   
4   bleu_frames\1039117223_frame_5.jpg   
5     bleu_frames\18337723_frame_1.jpg   
6     bleu_frames\18337723_frame_2.jpg   
7     bleu_frames\18337723_frame_3.jpg   
8     bleu_frames\18337723_frame_4.jpg   
9     bleu_frames\18337723_frame_5.jpg   
10     bleu_frames\3007804_frame_1.jpg   
11     bleu_frames\3007804_frame_2.jpg   
12     bleu_frames\3007804_frame_3.jpg   
13     bleu_frames\3007804_frame_4.jpg   
14     bleu_frames\3007804_frame_5.jpg   
15  bleu_frames\1031718032_frame_1.jpg   
16  bleu_frames\1031718032_frame_2.jpg   
17  bleu_frames\1031718032_frame_3.jpg   
18  bleu_frames\1031718032_frame_4.jpg   
19  bleu_frames\1031718032_frame_5.jpg   

                                          description  \
0   Miracle of nature - flowering oleanders in the...   
1   Miracle of nature - flowering oleanders in the...   
2   Miracle of nature - flowering oleanders in the...   
3   Miracle of nature - flowering oleanders in the...   
4   Miracle of nature - flowering oleanders in the...   
5               Aerial shot of a tropical rainforest.   
6               Aerial shot of a tropical rainforest.   
7               Aerial shot of a tropical rainforest.   
8               Aerial shot of a tropical rainforest.   
9               Aerial shot of a tropical rainforest.   
10                        Flames and black background   
11                        Flames and black background   
12                        Flames and black background   
13                        Flames and black background   
14                        Flames and black background   
15  Young juvenile and fish eggs of mahseer fish (...   
16  Young juvenile and fish eggs of mahseer fish (...   
17  Young juvenile and fish eggs of mahseer fish (...   
18  Young juvenile and fish eggs of mahseer fish (...   
19  Young juvenile and fish eggs of mahseer fish (...   

                                     blip_description  \
0                          pink flowers on olive tree   
1                          pink flowers on olive tree   
2                          pink flowers in the garden   
3                          pink flowers in the garden   
4                              pink flowers on a bush   
5            aerial view of the forest in the morning   
6   aerial drone video of a foggy forest in the mo...   
7   aerial drone video of a foggy forest in the mo...   
8   aerial view of the mountains and forest in the...   
9   aerial view of the mountains and forest in the...   
10                    fire flames on black background   
11                           fire flames in the night   
12                          a fire is seen in the sky   
13               a large fire is burning in the night   
14                           fire flames in the night   
15             a white wall with yellow flowers on it   
16          a white wall with a lot of yellow flowers   
17             a white wall with yellow flowers on it   
18          a white wall with a lot of yellow flowers   
19             a white wall with yellow flowers on it   

                                  summary_description  bleu_score  
0   Pink flowers are blooming on an olive tree, in...    0.018578  
1   Pink flowers are blooming on an olive tree, in...    0.018578  
2   Pink flowers are blooming on an olive tree, in...    0.018578  
3   Pink flowers are blooming on an olive tree, in...    0.018578  
4   Pink flowers are blooming on an olive tree, in...    0.018578  
5   A drone flies over a misty mountain forest in ...    0.015715  
6   A drone flies over a misty mountain forest in ...    0.015715  
7   A drone flies over a misty mountain forest in ...    0.015715  
8   A drone flies over a misty mountain forest in ...    0.015715  
9   A drone flies over a misty mountain forest in ...    0.015715  
1

In [14]:
import pandas as pd
from nltk.translate.bleu_score import sentence_bleu, SmoothingFunction
from tqdm import tqdm

# Load the CSV file
df = pd.read_csv('blip_pretrained_bleu_with_summaries.csv')

# Calculate BLEU scores with a progress bar
bleu_scores = []
smooth = SmoothingFunction().method4

for idx, row in tqdm(df.iterrows(), total=len(df), desc="Calculating BLEU Scores"):
    reference = row['description']
    candidate = row['summary_description']
    
    # Calculate BLEU score
    bleu_score = sentence_bleu([reference.split()], candidate.split(), smoothing_function=smooth)
    bleu_scores.append(bleu_score)

# Add the BLEU scores to the dataframe
df['bleu_score'] = bleu_scores

# Calculate the average BLEU score
average_bleu_score = sum(bleu_scores) / len(bleu_scores)

# Save the updated dataframe to a new CSV file
df.to_csv('blip_finetuned&summary_bleu_with_scores.csv', index=False)

# Print the BLEU scores and the average BLEU score
for idx, score in enumerate(bleu_scores):
    print(f"Frame {idx+1}: BLEU Score = {score:.4f}")

print(f"Average BLEU Score = {average_bleu_score:.4f}")

Calculating BLEU Scores: 100%|██████████| 495/495 [00:00<00:00, 5771.21it/s]

Frame 1: BLEU Score = 0.0186
Frame 2: BLEU Score = 0.0186
Frame 3: BLEU Score = 0.0186
Frame 4: BLEU Score = 0.0186
Frame 5: BLEU Score = 0.0186
Frame 6: BLEU Score = 0.0157
Frame 7: BLEU Score = 0.0157
Frame 8: BLEU Score = 0.0157
Frame 9: BLEU Score = 0.0157
Frame 10: BLEU Score = 0.0157
Frame 11: BLEU Score = 0.0000
Frame 12: BLEU Score = 0.0000
Frame 13: BLEU Score = 0.0000
Frame 14: BLEU Score = 0.0000
Frame 15: BLEU Score = 0.0000
Frame 16: BLEU Score = 0.0000
Frame 17: BLEU Score = 0.0000
Frame 18: BLEU Score = 0.0000
Frame 19: BLEU Score = 0.0000
Frame 20: BLEU Score = 0.0000
Frame 21: BLEU Score = 0.0209
Frame 22: BLEU Score = 0.0209
Frame 23: BLEU Score = 0.0209
Frame 24: BLEU Score = 0.0209
Frame 25: BLEU Score = 0.0209
Frame 26: BLEU Score = 0.0476
Frame 27: BLEU Score = 0.0476
Frame 28: BLEU Score = 0.0476
Frame 29: BLEU Score = 0.0476
Frame 30: BLEU Score = 0.0476
Frame 31: BLEU Score = 0.1645
Frame 32: BLEU Score = 0.1645
Frame 33: BLEU Score = 0.1645
Frame 34: BLEU Scor